api/search.rb

```diff
@@ -125,14 +125,21 @@ def get_threads(context, group_ids, local_params, search_text)
   result_obj.to_json
 end
 
+error Sinatra::Param::InvalidParameterError do
+  # NOTE (CCB): The current behavior of the service is to return a seemingly positive response
+  # for an invalid request. In the future the API's contract should be modified so that HTTP 400
+  # is returned. This informs the client that the request was invalid, rather than having to guess
+  # about an empty response body.
+  [200, '{}']
+end
+
 get "#{APIPREFIX}/search/threads" do
+  param :text, String, required: true
+  param :context, String, default: 'course'
+  param :sort_order, String, in: %w(asc desc), transform: :downcase
```

Comment:
@robrap Providing an illegal sort key in the past would have also resulted in returning an empty result.
The relevant lines are: https://github.com/edx/cs_comments_service/blob/master/lib/helpers.rb#L177-L179
https://github.com/edx/cs_comments_service/blob/master/lib/helpers.rb#L250-L261

It looks like for the other parameters (`text` and `context`), if they are malformed in some way, we would also get an empty result (as in, there would be an attempt to search on those parameters and return an empty result).